rct229/rulesets/ashrae9012019/section19/section19rule19.py

```diff
@@ -0,0 +1,262 @@
+                            "cooling_system",
+                            "cooling_system_type",
+                        )
+                        == COOLING_SYSTEM.NON_MECHANICAL
+                        for hvac_id_p in get_list_hvac_systems_associated_with_zone(
+                            rmi_p, zone_p["id"]
+                        )
+                    ]
+                )
+
+        return {
+            "applicable_hvac_sys_ids": applicable_hvac_sys_ids,
+            "zonal_exhaust_fan_elec_power_b": zonal_exhaust_fan_elec_power_b,
+            "zones_served_by_hvac_has_non_mech_cooling_bool_p": zones_served_by_hvac_has_non_mech_cooling_bool_p,
+            "zone_hvac_in_has_non_mech_cooling_p": zone_hvac_in_has_non_mech_cooling_p,
+        }
+
+    def list_filter(self, context_item, data):
+        applicable_hvac_sys_ids = data["applicable_hvac_sys_ids"]
+
+        return context_item.baseline["id"] in applicable_hvac_sys_ids
+
+    class HVACRule(RuleDefinitionBase):
+        def __init__(self):
+            super(Section19Rule19.HVACRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, True),
+                required_fields={
+                    "$": ["fan_system"],
+                },
+            )
+
+        def is_applicable(self, context, data=None):
+            hvac_b = context.baseline
+            hvac_id_b = hvac_b["id"]
+            applicable_hvac_sys_ids = data["applicable_hvac_sys_ids"]
+
+            return hvac_id_b in applicable_hvac_sys_ids
+
+        def get_calc_vals(self, context, data=None):
+            hvac_b = context.baseline
+
+            zonal_exhaust_fan_elec_power_b = data["zonal_exhaust_fan_elec_power_b"]
+            zones_served_by_hvac_has_non_mech_cooling_bool_p = data[
+                "zones_served_by_hvac_has_non_mech_cooling_bool_p"
+            ]
+            zone_hvac_in_has_non_mech_cooling_p = data[
+                "zone_hvac_in_has_non_mech_cooling_p"
+            ]
+
+            fan_sys_b = hvac_b["fan_system"]
+
+            fan_sys_info_b = (
+                get_fan_system_object_supply_return_exhaust_relief_total_power_flow(
+                    fan_sys_b
+                )
+            )
+
+            more_than_one_supply_fan_b = (
+                True if fan_sys_info_b["supply_fans_qty"] == 1 else False
+            )
+
+            total_fan_power_b = (
+                fan_sys_info_b["supply_fans_total_fan_power"]
+                + fan_sys_info_b["return_fans_total_fan_power"]
+                + fan_sys_info_b["exhaust_fans_total_fan_power"]
+                + fan_sys_info_b["relief_fans_total_fan_power"]
+                + zonal_exhaust_fan_elec_power_b
+            )
+
+            supply_fan_flow_b = fan_sys_info_b["supply_fans_airflow"]
+
+            fan_power_per_flow_b = (
+                total_fan_power_b / supply_fan_flow_b
+                if supply_fan_flow_b != ZERO.FLOW
+                else 0 * ureg("W/CFM")
```

Comment:
Usually we create a new ZERO in the ZERO class.
ZERO.POWER_PER_FLOW.

Reply:
Addressed.